Robot To Boiler Command Group

tfrancisco opened the pull request.

Commit:
fix build error

Commit:
merge w/ origin/master

Commit:
Merge remote-tracking branch 'origin/master' into tiona_vision

# Conflicts:
#	SeriouslyCommonLib

Commit:
point to private imask

Commit:
Move to hopper command group

Commit:
Move To Hopper Command with Rotate Robot to boiler command

Commit:
Reference version of common lib with checkstyle suppressions

Commit:
Merge branch 'vision' into tiona_vision

# Conflicts:
#	Competition2017/src/competition/subsystems/SubsystemDefaultCommandMap.java
#	Competition2017/src/competition/subsystems/vision/commands/RotateRobotToBoilerCommand.java
#	SeriouslyCommonLib

WasabiFan commented on Competition2017/src/competition/subsystems/autonomous/MoveToHopperCommandGroup.java:
```diff
@@ -0,0 +1,52 @@
+        this.addSequential(rotateToParallelCommand);
+
+        DriveForDistanceCommand driveAlongParallel = driveForDistanceProvider.get();
+        driveAlongParallel.setDeltaDistance(distanceFromBoiler.get() * Math.cos(angleToParallel.get()));
```
This value is only known when the command group starts running. Save the `DriveForDistanceCommand` instance to a private variable, and then in `initialize` for the command group you can compute the distance value and set it.

SethNguyenXBOT committed:
Added ShooterBelts to Smart Dashboard. (#62)

* Added ShooterBelts to Smart Dashboard.

* Changed "ShooterBelt" to "ShooterBelts"

* Added ShooterBelt intake and eject power log.

SethNguyenXBOT committed:
Feature/shoot fuel command (#60)

* Unfinished changes (2/16/2017).

* Created high-level "Shoot Fuel" command.

* Fixed Travis build errors.

* Fixed Travis Build errors.

* Created "ShootFuelCommandGroup" that runs both left and right fuel command groups.

* Made changes requested from Pull Request.

* Made changes requested from Pull Request. (reverted from commit 85616c905a746efa00b94c374ca8e717f661f587)

* Made changes from Pull Request.

Unagifan committed:
Feature/autonomous command selection (#61)

Created a autonomous command selector. Mapped the base autonomous command group/code to be activated by Smart DashBoard

JohnGilb committed:
Deal with XbotButton(s) build break

Unagifan committed:
Initial Oi Testing Outline

Mapped out all commands needed to actuate robot to oi or smart dashboard.

Unagifan committed:
Detailed End logging for Drive Commands

Created end logging for drive commands

Commit:
Basically A LOT of chanages made :)

Commit:
modify CalculateDistanceOfParallelCommand.java

Commit:
point to newer commonlib

Commit:
Merge remote-tracking branch 'origin/master' into tiona_vision

# Conflicts:
#	Competition2017/src/competition/operator_interface/OperatorCommandMap.java

Commit:
fix merge issues
